### ISSUE TITLE
date, time and datetime-local validation not working

### ISSUE THREAD

**Issue by germanbisurgi** [json-editor/json-editor#1471]
### Expected behavior

Empty strings should display error messages when using format date, time or datetime-local

### Actual behavior

No error message is display when using format date, time or datetime-local

### Steps to reproduce the behavior

[Direct link to example](https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUBXRBTA+vAhgLa75RQBOMARulFsihelgDQgTliHQ5VYBmYcrgAmWADZY6qJqxCZRA/OnFQcAN3zjmDWW0L4AHjjHQAFqgAMbGmQQ5EUUrkJgxqAIxsAxvmwOsJBhYdVwOSCxyKABPAPxybwtGcmY2RDMwAHccSM5yBhAYeDpyfG9YBBA2GG8EcWpUEEFi/Ez6LiwAVirwKgArLHKccXxosFpG+CEDcR7ERKwDVFBYKElGukcemIgsRrB+wage8N2omHplkBFnK529lBBHSngAcx7BcgNjx5vpAF82LBiHdortGs8iu82J9vhsYCDAddnMCsABacRgXyzNAge4QihQj7TUiNP5SBHozHYkD/OlAsyLB4gKhgMBQZ74CAAFlp/yAA==)

Add a remove values from inputs.

```javascript
{
  "title": "test",
  "type": "object",
  "properties": {
    "date": {
      "type": "string",
      "format": "date"
    },
    "time": {
      "type": "string",
      "format": "time"
    },
    "datetime-local": {
      "type": "string",
      "format": "datetime-local"
    }
  }
}
```

**Comment by germanbisurgi** [comment]
Fixed in #1472

**Comment by trycoon** [comment]
This change, introduced in version 2.14.x, should have been a major change ("3.00)"!
It broke our application completely. Please, make sure that when you make breaking changes, even if it is fixes what you consider being a bug, you should signal this change by bumping the major version. See https://semver.org/
